src/frontend/mosh-server.cc

```diff
@@ -783,6 +796,28 @@ static void serve( int host_fd, Terminal::Complete &terminal, ServerConnection &
 	    connected_utmp = true;
 	  }
 	  #endif
+      #ifdef HAVE_SYSLOG
+	  if ( syslog_saved_addr_len != network.get_remote_addr_len()
+	       || memcmp( &syslog_saved_addr, &network.get_remote_addr(),
+			  syslog_saved_addr_len ) != 0 ) {
+
+	    syslog_saved_addr = network.get_remote_addr();
+	    syslog_saved_addr_len = network.get_remote_addr_len();
+
+	    char host[ NI_MAXHOST ];
+	    int errcode = getnameinfo( &syslog_saved_addr.sa, syslog_saved_addr_len,
+				       host, sizeof( host ), NULL, 0,
+				       NI_NUMERICHOST );
+	    if ( errcode != 0 ) {
+	      throw NetworkException( std::string( "serve: getnameinfo: " ) + gai_strerror( errcode ), 0 );
+	    }
+        struct passwd *pw = getpwuid( getuid() );
+        if (pw == NULL) {
+	      throw NetworkException( std::string( "serve: getpwuid: " ) + gai_strerror( errno ), 0 );
```

Comment:
You want `strerror()` or `strerror_r()` here.

Reply:
Will fix this.